src/ESM.sol

```diff
@@ -70,12 +70,18 @@ contract ESM {
 
         end.cage();
         if (proxy != address(0)) {
-            VatLike(end.vat()).deny(proxy);
+            DenyLike(end.vat()).deny(proxy);
```

Comment:
I think we should test the path (it will work) where the `deny()` is called prior to the `fire()` and just ensure that works.  This will prevent us from introducing a regression in the future.  Other than that, I like the flexibility this gives the ESM.